src/renderer/apis/commands.ts

```diff
@@ -1,31 +1,251 @@
 import { Logger } from "../modules/logger";
+import { channels, constants, messages, users } from "../modules/common";
+import { getByStoreName } from "../modules/webpack";
 
-const commandsLogger = Logger.api("Commands");
+const logger = Logger.api("Commands");
 
-class CommandsAPI extends EventTarget {
-  public commands = new Map<string, RepluggedCommand>();
+interface CommandsAndSection {
+  section: RepluggedCommandSection;
+  commands: Map<string, AnyRepluggedCommand>;
+}
+
+export const commandAndSections = new Map<string, CommandsAndSection>();
 
-  public registerCommand(command: RepluggedCommand): void {
-    if (this.commands.has(command.name)) {
-      commandsLogger.error(`Command “${command.name}” is already registered!`);
-      return;
+export const defaultSection: RepluggedCommandSection = Object.freeze({
+  id: "replugged",
+  name: "Replugged",
+  type: 1,
+  icon: "https://cdn.discordapp.com/attachments/1000955992068079716/1004196106055454820/Replugged-Logo.png",
+});
+
+export class CommandInteraction<T extends CommandOptionReturn> {
+  public options: T[];
+  public channel: Channel;
+  public guild: Guild;
+  public constructor(props: { options: T[]; channel: Channel; guild: Guild }) {
+    const UploadAttachmentStore = getByStoreName<
+      Store & {
+        getUpload: (
+          channelId: string,
+          optionName: string,
+          draftType: 0,
+        ) => { uploadedFilename?: string; item?: { file: File } };
+      }
+    >("UploadAttachmentStore")!;
+    this.options = props.options;
+    this.channel = props.channel;
+    this.guild = props.guild;
+    for (const option of this.options.filter(
+      (o) => o.type === ApplicationCommandOptionType.Attachment,
+    )) {
+      const { uploadedFilename, item } =
+        UploadAttachmentStore.getUpload(props.channel.id, option.name, 0) ?? {};
+      option.value = { uploadedFilename, file: item?.file };
     }
+  }
 
-    this.commands.set(command.name, command);
-    this.dispatchEvent(new CustomEvent("rpCommandAdded", { detail: { command } }));
+  public getValue<K extends T["name"], D = undefined>(
+    name: K,
+    defaultValue?: D,
+  ): GetValueType<GetCommandOption<T, K>, D> {
+    return (this.options.find((o) => o.name === name)?.value ?? defaultValue) as GetValueType<
+      GetCommandOption<T, K>,
+      D
+    >;
   }
+}
+
+/**
+ * @internal
+ * @hidden
+ */
+async function executeCommand<T extends CommandOptions>(
+  cmdExecutor:
+    | ((
+        interaction: CommandInteraction<GetCommandOptions<T>>,
+      ) => Promise<RepluggedCommandResult> | RepluggedCommandResult)
+    | undefined,
+  args: Array<GetCommandOptions<T>>,
+  currentInfo: { guild: Guild; channel: Channel },
+  command: RepluggedCommand<T>,
+): Promise<void> {
+  try {
+    const currentChannelId = currentInfo.channel.id;
+    const loadingMessage = messages.createBotMessage({
+      channelId: currentChannelId,
+      content: "Executing Command, Please Wait...",
```

Comment:
Delete

Reply:
i left it there on purpose that if in future for some reason it doesnt show thinking message then it would show this.

Reply:
That makes no sense, and plus this is not i18n'd.

Reply:
well it's kinda backend fallback so no i18n but if it doesn't make sense then will remove it

Reply:
And I'm saying it doesn't make sense

Reply:
and I am saying I will delete it then